lib/AST/Decl.cpp

```diff
@@ -753,7 +748,15 @@ SourceRange Decl::getSourceRangeIncludingAttrs() const {
 }
 
 bool Decl::isInGeneratedBuffer() const {
-  return getModuleContext()->isInGeneratedBuffer(getStartLoc());
+  if (getModuleContext()->isInGeneratedBuffer(getStartLoc()))
+    return true;
+
+  // MissingDecl has a resolved decl if it was generated by a member attribute
+  // expansion.
+  if (auto *missing = dyn_cast<MissingDecl>(this))
+    return missing->getResolvedDecl();
```

Comment:
Is this necessary because the missing declaration has no source location? Would it be reasonable to instead give the missing declaration we create in the parser a source location (say, at the end of the generated file) so we elide this check?